src/main/kotlin/com/terraformation/backend/admin/AdminParticipantsController.kt

```diff
@@ -227,6 +230,41 @@ class AdminParticipantsController(
     return redirectToParticipant(participantId)
   }
 
+  @PostMapping("/updateProjectDealNames")
+  fun updateProjectDealNames(): String {
+    val participants = participantStore.findAll()
+    participants.forEach { participant ->
+      val projectIds = participant.projectIds
+      if (projectIds.size == 1) {
+        projectAcceleratorDetailsService.update(projectIds.single()) {
+          it.copy(dealName = participant.name)
+        }
+      } else {
+        projectIds.forEachIndexed { index, projectId ->
+          if (index == 0) {
+            projectAcceleratorDetailsService.update(projectId) {
+              it.copy(dealName = participant.name)
+            }
+          } else {
+            projectAcceleratorDetailsService.update(projectId) {
+              it.copy(dealName = "${participant.name} ${index + 1}")
+            }
+          }
+        }
+      }
+    }
+
+    val applications = applicationStore.fetchAll()
+    applications.forEach {
+      projectAcceleratorDetailsService.update(it.projectId) { existing ->
+        // Update only projects that did not have a deal name already.
+        existing.copy(dealName = existing.dealName ?: it.internalName)
```

Comment:
This would be clearer if the outer `forEach` declared a parameter name; I had to think about this a little to figure out that `it` referred to the lambda argument from a parent scope rather than the innermost one.